presto-main/src/test/java/io/prestosql/operator/TestDynamicFilterSourceOperator.java

```diff
@@ -277,63 +394,105 @@ public void testCollectEmptyBuildSide()
-                channel(1, VARCHAR));
-        verifyPassthrough(createOperator(operatorFactory),
-                ImmutableList.of(VARCHAR, VARCHAR),
-                largePage);
-        operatorFactory.noMoreOperators();
-        assertEquals(partitions.build(), ImmutableList.of(TupleDomain.all()));
+        List<TupleDomain<DynamicFilterId>> expectedTupleDomains = ImmutableList.of(
+                TupleDomain.withColumnDomains(ImmutableMap.of(
+                        new DynamicFilterId("0"), Domain.create(ValueSet.ofRanges(
+                                range(VARCHAR, utf8Slice(largeTextA), true, utf8Slice(largeTextA), true)), false),
+                        new DynamicFilterId("1"), Domain.create(ValueSet.ofRanges(
+                                range(VARCHAR, utf8Slice(largeTextB), true, utf8Slice(largeTextB), true)), false))));
+        assertDynamicFilters(100, maxSize, 100, expectedTupleDomains, ImmutableList.of(VARCHAR, VARCHAR), largePage);
+    }
+
+    @Test
+    public void testCollectMultipleLargePages()
+    {
+        int maxPositionsCount = 100;
+        Page page1 = new Page(createLongSequenceBlock(maxPositionsCount / 2, maxPositionsCount + (maxPositionsCount / 2) + 1));
+        Page page2 = new Page(createLongSequenceBlock(0, maxPositionsCount + 1));
+        Page page3 = new Page(createLongSequenceBlock(maxPositionsCount, (2 * maxPositionsCount) + 1));
+
+        TupleDomain<DynamicFilterId> expectedTupleDomain = TupleDomain.withColumnDomains(ImmutableMap.of(
+                        new DynamicFilterId("0"), Domain.create(ValueSet.ofRanges(
+                                range(BIGINT, 0L, true, 2L * maxPositionsCount, true)), false)));
+        assertDynamicFilters(maxPositionsCount, ImmutableList.of(expectedTupleDomain), ImmutableList.of(BIGINT), page1, page2, page3);
     }
 
     @Test
     public void testCollectDeduplication()
     {
-        int maxRowCount = getDynamicFilteringMaxPerDriverRowCount(pipelineContext.getSession());
-        Page largePage = new Page(createLongRepeatBlock(7, maxRowCount * 10)); // lots of zeros
-        Page nullsPage = new Page(createLongsBlock(Arrays.asList(new Long[maxRowCount * 10]))); // lots of nulls
+        int maxPositionsCount = 100;
+        Page largePage = new Page(createLongRepeatBlock(7, maxPositionsCount * 10)); // lots of zeros
+        Page nullsPage = new Page(createLongsBlock(Arrays.asList(new Long[maxPositionsCount * 10]))); // lots of nulls
 
-        OperatorFactory operatorFactory = createOperatorFactory(channel(0, BIGINT));
-        verifyPassthrough(createOperator(operatorFactory),
+        TupleDomain<DynamicFilterId> expectedTupleDomain = TupleDomain.withColumnDomains(ImmutableMap.of(
+                        new DynamicFilterId("0"), Domain.create(ValueSet.of(BIGINT, 7L), false)));
+        assertDynamicFilters(maxPositionsCount, ImmutableList.of(expectedTupleDomain), ImmutableList.of(BIGINT), largePage, nullsPage);
+    }
+
+    @Test
```

Comment:
add a test case for min/max when one of the columns has nulls only while the other one has values from 0 to `maxPositionsCount+1`

Reply:
Added `testMultipleColumnsCollectMinMaxWithNulls`